.teamcity/common/extensions.kt

```diff
@@ -102,6 +102,10 @@ fun BuildSteps.checkCleanM2(os: Os = Os.linux) {
 
 fun buildToolGradleParameters(daemon: Boolean = true, isContinue: Boolean = true, os: Os = Os.linux): List<String> =
     listOf(
+        // We pass the 'maxParallelForks' setting as 'workers.max' to limit the maximum number of executers even
+        // if multiple test tasks run in paralled. We also pass it to the Gradle build as a maximum (PmaxParallelForks)
+        // for each test task as well, such that we are independent of whatever default value is defined in the build itself.
+        "-Dorg.gradle.workers.max=%maxParallelForks%"
```

Comment:
I wonder what the effect on performance of the build is by reducing the number of parallel workers. Apart from running the tests. I suppose it doesn't have a big effect. WDYT?

Reply:
As discusses offline, we are doing the change and observe.